### PR TITLE
Admin panel base

### PR DESCRIPTION
**Pull request by jniemii**
This PR creates the base for adminJS for the project. It adds a user role for all users, which are used to differentiate between normal users and admins. Currently users can only be elevated to admins through ssh. The admin panel is protected behind the admin user login. Currently viewing, deleting and creating users and changing their passwords are implemented. Admins and superadmins are differentiated in the database but currently have identical functionality.

NOTE: everyone will have to copy .env  from .env-sample after this is merged.

### CODE REVIEW

File: backend/src/routers/index.ts [hunk 1]
```diff
@@ -3,4 +3,5 @@ export { exportRouter } from './export';
 export { healthRouter } from './health';
 export { userRouter } from './user';
 export { presetsRouter } from './presets';
+export { getAdminRouter } from './admin';
```
Comment (hmalmsten) [inline backend/src/routers/index.ts:6]: Minor thing, but just "adminRouter" would be more consistent with the other exports

**Review by hmalmsten — APPROVED** [review]

File: backend/tsconfig.json [hunk 2]
```diff
@@ -27,7 +27,7 @@
         /* Modules */
         "module": "commonjs" /* Specify what module code is generated. */,
```
Comment (KorpelainenTommi) [inline backend/tsconfig.json]: Seems to be an accidental change, inconsistent comment whitespace REEE

File: backend/src/routers/admin.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    }
+
+    const role = await selectUserRole(user.message);
+    if (role == 'admin' || role == 'superadmin') {
+        return Promise.resolve({ email, password });
+    }
+
+    return null;
+};
+
```
Comment (KorpelainenTommi) [inline backend/src/routers/admin.ts:47]: minor typo

**Review by KorpelainenTommi — COMMENTED** [review]
Minor typos

File: backend/tsconfig.json
Comment (KorpelainenTommi) [inline backend/tsconfig.json:1]: Looks like an accidental change

File: backend/src/routers/admin.ts [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+const getSesssionSecret = (): string => {
```
Comment (KorpelainenTommi) [inline backend/src/routers/admin.ts]: Minor typo

**Review by KorpelainenTommi — APPROVED** [review]

File: backend/src/routers/index.ts (hunk 1 above)
Comment (jniemii) [reply backend/src/routers/index.ts:6]: Unlike other routers it's behind an async function you have to await so I think "get" makes sense for it.

**Pull request merged**